app/models/manageiq/providers/azure/cloud_manager.rb

```diff
@@ -12,12 +12,40 @@ class ManageIQ::Providers::Azure::CloudManager < ManageIQ::Providers::CloudManag
   require_nested :ProvisionWorkflow
   require_nested :OrchestrationStack
   require_nested :OrchestrationServiceOptionConverter
-  require_nested :SecurityGroup
+
+  include ManageIQ::Providers::Azure::ManagerMixin
 
   alias_attribute :azure_tenant_id, :uid_ems
 
   has_many :resource_groups, :foreign_key => :ems_id, :dependent => :destroy
 
+  has_one :network_manager,
+          :foreign_key => :parent_ems_id,
+          :class_name  => "ManageIQ::Providers::Azure::NetworkManager",
+          :autosave    => true,
+          :dependent   => :destroy
+
+  delegate :floating_ips,
+           :security_groups,
+           :cloud_networks,
+           :cloud_subnets,
+           :network_ports,
+           :network_routers,
+           :public_networks,
+           :private_networks,
+           :all_cloud_networks,
+           :to        => :network_manager,
+           :allow_nil => true
+
+  before_validation :ensure_managers
+
+  def ensure_managers
+    build_network_manager unless network_manager
+    network_manager.name            = "#{name} Network Manager"
+    network_manager.zone_id         = zone_id
+    network_manager.provider_region = provider_region
+  end
+
```

Comment:
@bronaghs yeah I guess we can refactor this in a next PR. I think this code might stay, even when we will be switching to another NetworkManager